tools/distpartitioning/data_shuffle.py

```diff
@@ -480,14 +547,16 @@ def gen_dist_partitions(rank, world_size, params):
     print('[Rank: ', rank, '] Done assigning global-ids to nodes...')
 
     #shuffle node feature according to the node order on each rank. 
-    for ntype_name in ntypes: 
-        if (ntype_name+'/feat' in rcvd_global_nids):
-            global_nids = rcvd_global_nids[ntype_name+'/feat']
-
-            common, idx1, idx2 = np.intersect1d(node_data[constants.GLOBAL_NID], global_nids, return_indices=True)
-            shuffle_global_ids = node_data[constants.SHUFFLE_GLOBAL_NID][idx1]
-            feature_idx = shuffle_global_ids.argsort()
-            rcvd_node_features[ntype_name+'/feat'] = rcvd_node_features[ntype_name+'/feat'][feature_idx]
+    for ntype_name in ntypes:
+        featnames = get_ntype_featnames(ntype_name, schema_map)
+        for featname in featnames:
+            if (ntype_name+'/'+featname in rcvd_global_nids):
```

Comment:
why a node feature doesn't exist?

Reply:
if we know it exists for sure, we need to use assert instead of "if"

Reply:
Agreed. Changed the code appropriately. This same logic was elsewhere as well and it was mentioned in one of the earlier code review comments.